pkg/storage/aws/s3.go

```diff
@@ -17,27 +18,25 @@ func BucketRegionIsDiscoverable(bucket string) bool {
 // GetBucketRegion returns the AWS region that a bucket is in, or an error
 // if the region cannot be determined.
 // copied from https://github.com/openshift/openshift-velero-plugin/pull/223/files#diff-da482ef606b3938b09ae46990a60eb0ad49ebfb4885eb1af327d90f215bf58b1
+// modified to aws-sdk-go-v2
 func GetBucketRegion(bucket string) (string, error) {
 	var region string
-
-	sessionInstance, err := session.NewSession()
+	// GetBucketRegion will attempt to get the region for a bucket using the client's configured region to determine
+	// which AWS partition to perform the query on.
+	// Client therefore needs to be configured with region.
+	// In local dev environments, you might have ~/.aws/config that could be loaded and set with default region.
+	// In cluster/CI environment, ~/.aws/config may not be configured, so set region explicitly.
+	// Also set to use anonymous credentials
+	cfg, err := config.LoadDefaultConfig(context.TODO(),
+		config.WithRegion("us-east-1"),
+		config.WithCredentialsProvider(aws.AnonymousCredentials{}),
+	)
 	if err != nil {
-		// return "", errors.WithStack(err) // don't need stack trace, not inside velero-plugin
 		return "", err
 	}
-
-	for _, partition := range endpoints.DefaultPartitions() {
-		for regionHint := range partition.Regions() {
-			region, _ = s3manager.GetBucketRegion(context.Background(), sessionInstance, bucket, regionHint)
-
-			// we only need to try a single region hint per partition, so break after the first
-			break
-		}
-
-		if region != "" {
-			return region, nil
-		}
+	region, err = manager.GetBucketRegion(context.Background(), s3.NewFromConfig(cfg), bucket)
```

Comment:
Don't we want to make this call first and then fall back if not gathered to default region?

Reply:
This call will be made to a specific region hinting server. It cannot be called without specifying a specific region before hand.